### PR TITLE
Refactor Health and Version, adapt to jicoco.

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: src/main/java/org/jitsi/jicofo/Main.java [hunk 1]
```diff
@@ -62,17 +64,12 @@ public static void main(String[] args)
...
-            synchronized (exitSyncRoot)
-            {
-                exitSyncRoot.notifyAll();
-            }
-        }));
+        Runtime.getRuntime().addShutdownHook(new Thread(shutdownService::beginShutdown));
         logger.info("Starting OSGi services.");
-        BundleActivator activator = startOsgi(exitSyncRoot);
+        BundleActivator activator = new DummyActivator();
+        OSGiLauncher launcher = startOsgi(activator);
```
Comment (bbaldino) [inline src/main/java/org/jitsi/jicofo/Main.java:72]: We have to still start OSGi for some things, right?  But the dummy activator does nothing, so what behavior is it we're relying on there?

Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/Main.java:72]: At first flance `OSGiLauncher.start(BundleActivator)` requires a non-null activator, so I added that without digging further. But I see there a `start()` with no arguments that I can use.

Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/Main.java:72]: Ugh, `start()` is private...

Comment (bbaldino) [reply src/main/java/org/jitsi/jicofo/Main.java:72]: Ok...but what is it that happens as part of `OSGiLauncher.start` that we still need? (Just curious)

Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/Main.java:72]: Starting all of our bundles, we still have a bunch:
https://github.com/jitsi/jicofo/blob/master/src/main/java/org/jitsi/jicofo/osgi/JicofoBundleConfig.java#L69

Comment (bbaldino) [reply src/main/java/org/jitsi/jicofo/Main.java:72]: Oh, and those still work even though the activator is a no-op?

Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/Main.java:72]: Yep 🤷

Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/Main.java:72]: I think I could move `WaitableBundleActivator` from the bundle config here, but I'd rather not fiddle with it

**Review by bbaldino — DISMISSED** [review]

**Review by bbaldino — APPROVED** [review]

**Pull request merged**